lib/propmangle.js

```diff
@@ -159,7 +162,7 @@ function mangle_properties(ast, options) {
         if (options.only_cache) {
             return cache.props.has(name);
         }
-        if (/^[0-9.]+$/.test(name)) return false;
+        if (/^-?[0-9]+(\.[0-9]+)?(e[+-][0-9]+)?$/.test(name)) return false;
```

Comment:
> The return value of  function as_property_name  is a string in all cases, correct?

Nope - see https://github.com/mishoo/UglifyJS2/pull/1779#discussion_r109491763

It's the raw value of whatever `AST_Constant` is storing.